company-template.el

```diff
@@ -55,19 +56,44 @@
 
 (defun company-template-forward-field ()
   (interactive)
+  (let ((start (point))
+        (next-field-start (company-template-get-next-field-start)))
+    (push-mark)
+    (goto-char next-field-start)
+    (company-template-remove-field (company-template-field-at start))))
+
+(defun company-template-clear-field ()
+  "Clear field at point.
+If there is a next field, delete everything up to next field.  If field at
+point is last field, remove preceeding comma if present."
+  (interactive)
   (let* ((start (point))
-         (templates (company-template-templates-at (point)))
+         (current-ovl (company-template-field-at start)))
+    (when current-ovl
+      (let ((this-field-end (overlay-end current-ovl))
+            (next-field-start (company-template-get-next-field-start)))
+        (cond ((and (company-template-field-at next-field-start)
+                    (< this-field-end next-field-start)
+                    (string-match "^\\s-*,+\\s-*$" (buffer-substring-no-properties
+                                                    this-field-end next-field-start)))
+               (delete-region start next-field-start))
+              ((and (not (company-template-templates-at
+                          (company-template-get-next-field-start)))
+                    (looking-back ",+\\s-*" nil t))
```

Comment:
The second argument shouldn't be nil, let's use `(line-beginning-position)` instead.